### PR TITLE
FRURLProtocol ignores ssl pinning

### PR DESCRIPTION
**Pull request by george-bafaloukas-forgerock**
# JIRA Ticket

Please, link jira ticket here.](https://bugster.forgerock.org/jira/browse/SDKS-3239)

# Description

Fix for SSL Pinning and FRURLProtocol uses

### CODE REVIEW

**Comment by jeyanthanperiyasamy** [comment]
can we add some tests around this for both success and failure cases?

**Review by jeyanthanperiyasamy — CHANGES_REQUESTED** [review]
comments provided

**Comment by george-bafaloukas-forgerock** [comment]
> can we add some tests around this for both success and failure cases?

Might a bit hard to get proper success for SSL Pinning, unless we set on the tests the keyhashes of the test server

File: FRAuth/FRAuth/Network/FRURLProtocol.swift [hunk 1]
```diff
@@ -20,6 +21,8 @@ import Foundation
     /// AuthorizationPolicy for URLProtocol
     @objc public static var authorizationPolicy: AuthorizationPolicy?
     
+    @objc public static var frSecurityConfiguration: FRSecurityConfiguration?
```
Comment (rodrigoareis) [inline FRAuth/FRAuth/Network/FRURLProtocol.swift:24]: Missing doc

**Review by rodrigoareis — CHANGES_REQUESTED** [review]
Overall changes looks good. Just minor comment to address

**Comment by george-bafaloukas-forgerock** [comment]
> Overall looks good.
> 
> Can you please explain why we need this part (I see you added it later) else { completionHandler(.performDefaultHandling, nil) return } we didn't have this before adding the ssl pinning part.

Yes I had to add this, as I have added the 
```func urlSession(_ session: URLSession, task: URLSessionTask, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void)```
Method and on some requests it might return other types of `authenticationMethod`. We need to allow `defaultHandling` for the rest otherwise this will not call the `completion` and will be an API misuse. This was found through the test

**Review by vahancouver — APPROVED** [review]
LGTM

**Review by rodrigoareis — APPROVED** [review]
Looks good to me

**Pull request merged**